File: Python/String/E-9. Palindrome Number.py
class Solution:
    @classmethod
    def isPalindrome(self, x: int) -> bool:
        reverse = 0
        orig_x = x

        while x > 0:
            reverse = reverse * 10 + x % 10
            x //= 10

        return reverse == orig_x


test_data = [
    {
        "x": 121,
        "ans": True,
    },
    {
        "x": -121,
        "ans": False,
    },
    {
        "x": 10,
        "ans": False,
    },
]

for data in test_data:
    print(data["ans"] == Solution.isPalindrome(data["x"]))

File: Javascript/String/E-9. Palindrome Number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  // return x.toString() === x.toString().split('').reverse().join('');

  let reverse = 0;
  const origX = x;

  while (x > 0) {
    reverse = reverse * 10 + (x % 10);
    x = parseInt(x / 10);
  }

  return reverse === origX;
};

const testData = [
  {
    x: 121,
    ans: true,
  },
  {
    x: -121,
    ans: false,
  },
  {
    x: 10,
    ans: false,
  },
];

for (const data of testData) {
  console.log(data.ans === isPalindrome(data.x));
}
